### PR TITLE
Add context menu to activity list

### PR DESCRIPTION
**Pull request by TheOneRing**
image

### CODE REVIEW

**Comment by TheOneRing** [comment]
Depends on https://github.com/owncloud/client/pull/8584

File: src/gui/activitywidget.cpp [hunk 1]
```diff
@@ -174,55 +173,6 @@ void ActivityWidget::slotAccountActivityStatus(AccountState *ast, int statusCode
     showLabels();
 }
 
-// FIXME: Reused from protocol widget. Move over to utilities.
-QString ActivityWidget::timeString(QDateTime dt, QLocale::FormatType format) const
```
Comment (fmoc) [inline src/gui/activitywidget.cpp:178]: This change appears to be unrelated cleanup and should be in a separate commit. Otherwise, the Git history becomes quite useless.

File: src/gui/protocolwidget.cpp [hunk 2]
```diff
@@ -129,7 +131,6 @@ void ProtocolWidget::showContextMenu(QWidget *parent, ProtocolItemModel *model,
 
 void ProtocolWidget::slotItemContextMenu()
 {
-    QModelIndexList list;
```
Comment (fmoc) [inline src/gui/protocolwidget.cpp:132]: This seems like some unrelated cleanup as well.

File: src/gui/activitywidget.cpp [hunk 3]
```diff
@@ -313,24 +253,23 @@ void ActivityWidget::slotBuildNotificationDisplay(const ActivityList &list)
 
     const auto accId = list.first().uuid();
     QList<Activity::Identifier> strayCats;
-    for (const auto &id : _widgetForNotifId.keys()) {
-        NotificationWidget *widget = _widgetForNotifId[id];
+    for (auto it = _widgetForNotifId.cbegin(); it != _widgetForNotifId.cend(); ++it) {
```
Comment (fmoc) [inline src/gui/activitywidget.cpp:256]: Is this change somehow related to the topic of the PR? I don't see the relation between using an iterator here and the new context menu.

**Review by fmoc — APPROVED** [review]

File: src/gui/activitywidget.cpp (hunk 1 above)
Comment (TheOneRing) [reply src/gui/activitywidget.cpp:178]: I just removed custom date handling, we just provide a qdate to the view and it takes care of the presentation.
As we no longer sort by the display string we don't need to have seconds in the display string.
